index.bs

```diff
@@ -2661,6 +2688,99 @@ create them does not matter.
+ 1. If |signal| is not undefined,
+  1. Let |abortAlgorithm| be the following steps:
+   1. Let |error| be |signal|'s [=AbortSignal/abort reason=].
+   1. Run |errorSteps| with |error|.
+   1. Let |actions| be an empty [=ordered set=].
+   1. If |stream|.[=ReadableStream/[[state]]=] is "`readable`", [=set/append=] the following action action to |actions|:
+     1. return ! [$ReadableStreamCancel$](|stream|, |error|).
+     1. Otherwise, return [=a promise resolved with=] undefined.
+   1. [=Shutdown with an action=] consisting of [=getting a promise to wait for all=] of the actions
+      in |actions|, and with |error|.
+  1. If |signal| is [=AbortSignal/aborted=], perform the following steps:
+    1. perform |abortAlgorithm|.
+    1. return |promise|.
+  1. [=AbortSignal/Add=] |abortAlgorithm| to |signal|.
+ 1. [=fully read=] |stream| given |successSteps|, |errorSteps|, and |stream|’s relevant global object.
+ 1. Return promise.
+</div>
+
+<div algorithm>
+ <dfn abstract-op lt="ReadableStreamConsumeAsText" id="ReadableStreamConsumeAsText">ReadableStreamConsumeAsText(|stream|, |options|})</dfn> will
+ consume all bytes produced by a given readable stream and return a promise that resolves with a string containing the UTF-8-decoded text.
+
+ It performs the following steps:
+
+ 1. Assert: |stream| [=implements=] {{ReadableStream}}.
+ 1. If ! [$IsReadableStreamLocked$]([=stream=]) is true, return [=a promise rejected with=] a
+    {{TypeError}} exception.
+ 1. If ! |stream|.[=ReadableStream/[[disturbed]]=] is true, return [=a promise rejected with=] a
+    {{TypeError}} exception.
+ 1. Let |signal| be |options|["{{ConsumeBytesOptions/signal}}"] if it [=map/exists=], or undefined
+    otherwise.
+ 1. Assert: either |signal| is undefined, or |signal| [=implements=] {{AbortSignal}}.
+ 1. Return the result of running [=consume fully with abort=] with |stream|, |signal|, and <a>UTF-8 decode</a>.
+</div>
+
+<div algorithm>
+ <dfn abstract-op lt="ReadableStreamConsumeAsBytes" id="ReadableStreamConsumeAsBytes">ReadableStreamConsumeAsBytes(|stream|, |options|})</dfn> will
+ consume all bytes produced by a given readable stream and return a promise that resolves with a single Uint8Array containing the read bytes.
+
+ It performs the following steps:
+
+ 1. Assert: |stream| [=implements=] {{ReadableStream}}.
+ 1. If ! [$IsReadableStreamLocked$]([=stream=]) is true, return [=a promise rejected with=] a
+    {{TypeError}} exception.
+ 1. If ! |stream|.[=ReadableStream/[[disturbed]]=] is true, return [=a promise rejected with=] a
+    {{TypeError}} exception.
+ 1. Let |signal| be |options|["{{ConsumeBytesOptions/signal}}"] if it [=map/exists=], or undefined
+    otherwise.
+ 1. Assert: either |signal| is undefined, or |signal| [=implements=] {{AbortSignal}}.
+ 1. Let |decode as Uint8Array| be an algorithm that takes a [=byte sequence=] |bytes| and runs the steps:
+  1. Let |arrayBuffer| be a new {{ArrayBuffer}} whose contents are |bytes|.
```

Comment:
This is a faithful copy of the text in fetch, but fetch [has a bug](https://github.com/whatwg/fetch/issues/1675) which is copied here. It doesn't necessarily need to be fixed here, just gotta make sure it gets fixed here as well when fixed upstream.